test_shopinvader_payment/tests/test_quotation.py

```diff
@@ -22,40 +21,47 @@ def test_get_no_cart_payment_info(self):
 class ShopinvaderQuotationPaymentCase(CommonConnectedQuotationCase):
     def setUp(self, *args, **kwargs):
         super(ShopinvaderQuotationPaymentCase, self).setUp(*args, **kwargs)
-        self.account_payment_mode = self.env.ref(
-            "shopinvader_payment.payment_method_check"
-        )
-        self.cart.write({"payment_mode_id": self.account_payment_mode.id})
+        self.payment_mode = self.env.ref(
+            "shopinvader_payment_manual.shopinvader_payment_check"
+        ).payment_mode_id
+        with self.work_on_services(
+            partner=self.partner, shopinvader_session=self.shopinvader_session
+        ) as work:
+            self.payment_service = work.component(usage="payment_manual")
 
     def test_get_cart_payment_info(self):
-        self.cart.order_line[0].product_id.only_quotation = True
-        response = self.service.dispatch("search", params={"id": self.cart.id})
+        self.quotation.order_line[0].product_id.only_quotation = True
+        response = self.service.dispatch(
```

Comment:
You don't test the quoration service... Am I wrong?

Reply:
self.service is the cart service

Reply:
self.service here is quotations service https://github.com/shopinvader/odoo-shopinvader/blob/12.0/shopinvader_quotation/tests/test_quotation.py#L40
Maybe we should put the name explicitly

Reply:
@sebastienbeau I miss something... what's the purpose of the quotation service? When do you use the quotation service or the car service.I can't get a clear conceptual representation between cart and quotation service

Reply:
When you have a cart you can pay it or request a quotation. (And in the case that you have a product that required quotation, you can not pay it).
When requesting a quotation on the cart https://github.com/shopinvader/odoo-shopinvader/blob/12.0/shopinvader_quotation/services/cart.py#L12. The typology of the cart change and it's set to "quotation".

So now this is not a cart anymore, it's a quotation in the state "estimating" https://github.com/shopinvader/odoo-shopinvader/blob/12.0/shopinvader_quotation/models/sale.py#L13

Odoo user will update the price of the quotation and click on "sent by email". state => estimated

Now the quotation can be accept and paid by the user using a payment method

Reply:
@lmignon tell me if you need more info

Reply:
I am going to clean improve the test. I reuse existing from 10. But method name are not really explicit